pkg/build/controller/common/util.go

```diff
@@ -42,6 +48,85 @@ func HandleBuildCompletion(build *buildapi.Build, runPolicies []policy.RunPolicy
 	if err := runPolicy.OnComplete(build); err != nil {
 		utilruntime.HandleError(fmt.Errorf("failed to run policy on completed build %s/%s: %v", build.Namespace, build.Name, err))
 	}
+	if err := HandleBuildPruning(build, buildLister, buildConfigGetter, buildDeleter); err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to prune old builds %s/%s: %v", build.Namespace, build.Name, err))
+	}
+}
+
+func isCompletedBuild(build buildapi.Build) bool {
+	return build.Status.Phase == buildapi.BuildPhaseComplete
+}
+
+func isFailedBuild(build buildapi.Build) bool {
+	return build.Status.Phase == buildapi.BuildPhaseFailed
+}
```

Comment:
Note that there is an existing IsBuildComplete helper which checks if the build is in any terminal state.  you might want to rename it to IsBuildTerminated or change it to IsBuildActive and reverse all the boolean checks that were calling the old function.